skycoin/skywire
# Updating mechanism for mainnet

NOTE: After merging this PR, all `git` tags and GitHub releases should be removed because they exist for testing purposes.

Did you run `make format && make check`? yes

Fixes #129
Fixes #141

It should be merged simultaneously with https://github.com/SkycoinPro/skywire-services/pull/84.

 Changes:	
- Implement `skywire-visor` and `skywire-cli` binary files update via the `hypervisor`
- Change `GET /visors/{pk}/restart` to `POST /visors/{pk}/restart`
- Setup `goreleaser` for `skywire-visor` and `skywire-cli`

How to test this PR:
1. Comment out the `VERSION := $(shell git describe)` line in `Makefile`
2. Add a new line below it: `VERSION := 0.1.0`
3. Run the generic integration environment
4. Go to `VisorA` pane
5. Make an HTTP `POST` request to `localhost:8080/api/visors/024ec47420176680816e0406250e7156465e4531f5b26057c9f6297bb0303558c7/update`
6. Check if there are no errors in `VisorA` pane and `VisorA` is restarted successfully and works. Apps should also work. Versions of all binaries in their output should be updated.
7. Check if the HTTP response status is `200` and body is `true`
8. Go to `shell` pane
9. Check if `skywire-visor`, `skywire-cli` and `VisorA` apps binaries are updated and work (e.g. using `ls -lah ./bin`)
10. Check if files/directories from logs from `VisorA` saying `Removing file ...` are removed.
11. Note that in fact, an old version of binaries is uploaded to the last GitHub release. GitHub releases need a clean-up.
